src/java.base/share/classes/java/io/ObjectInputStream.java

```diff
@@ -188,15 +190,33 @@
  * protected) or that there are get and set methods that can be used to restore
  * the state.
  *
- * <p>The contents of the stream can be filtered during deserialization.
- * If a {@linkplain #setObjectInputFilter(ObjectInputFilter) filter is set}
- * on an ObjectInputStream, the {@link ObjectInputFilter} can check that
+ * The deserialization filter for a stream is determined in one of the following ways:
+ * <ul>
+ * <li>A JVM-wide filter factory can be set via {@link Config#setSerialFilterFactory(BinaryOperator)}
+ *     or the system property {@code jdk.serialFilterFactory} or
+ *     the security property {@code jdk.serialFilterFactory}.
+ *     The filter factory is invoked for each new ObjectInputStream and
+ *     when a filter is set for a stream.
+ *     The filter factory determines the filter to be used for each stream based
+ *     on its inputs, thread context, other filters, or state that is available.
```

Comment:
Maybe a link to the ObjectInputFilter API documentation where it is explained what the two filters passed to the factory are in each of these cases should be provided here.

Namely: 

- in the constructor, `factory.apply(null, Config.getSerialFilter())` is invoked.
- in `setObjectInputFilter(newfilter)`, `factory.apply(filter, newFilter)` is invoked - where `filter` is the filter that the stream is currently using.

Or maybe link to the constructor and setObjectInputFilter method where this is explained.